live/live-root/usr/lib/dracut/modules.d/99agama-dud/agama-dud-apply.sh

```diff
@@ -156,6 +157,98 @@ create_repo() {
   "$NEWROOT/usr/bin/chroot" "$NEWROOT" createrepo_c "${repo_dir##"$NEWROOT"}"
 }
 
+# Finds the kernel modules to update
+#
+# It searches for the modules in the modules/ directory of the update.
+find_kernel_modules() {
+  local directory=$1
+  local -n modules=$2
+  local module_name
+  local files
+
+  modules=()
+  files=("${directory}"/*.ko*)
+  for module in "${files[@]}"; do
+    module_name=${module#"${directory}/"}
+    module_name=${module_name%.ko*}
+
+    if [[ ! " ${modules[*]} " =~ \ ${module_name}\  ]]; then
+      modules+=("$module_name")
+    fi
+  done
+
+  echo "Found ${#files[@]} kernel modules"
+}
+
+# Copies a kernel module
+#
+# It searches for a module with the same name. If found, it replaces it.
+# Otherwise, it copies the module to the top-level modules directory.
+copy_kernel_module() {
+  local source_dir=$1
+  local module=$2
+  local target_dir=$3
+  local source_file
+
+  echo "Copying ${module}..."
+
+  # expect a single file with $module.ko* name
+  source_file=("${source_dir}/${module}".ko*)
+
+  echo old "${target_dir}/**/*/${module}.ko*"
+  old_module=("${target_dir}"/**/*/"${module}".ko*)
+  if [ "${#old_module[@]}" -eq "1" ]; then
+    info "  Replacing the module ${old_module[0]}"
+    cp "${source_file[0]}" "${old_module[0]}"
+  elif [ "${#old_module[@]}" -eq "0" ]; then
+    info "  Not found the module to replace, so copying to kernel/ directory."
+    cp "${source_file[0]}" "${target_dir}"
+  else
+    info "  Skipping the module because several modules with the same name were found."
+  fi
+}
+
+# Updates kernel modules
+#
+# It copies the kernel modules and the hardware information for hwinfo.
```

Comment:
I do not get part about hardware information.

Reply:
Another left-over.